### DISCUSSION TITLE
Override nvim-cmp sources options and luasnip from_vscode

### DISCUSSION THREAD

**Discussion by mstinsky** [LazyVim/LazyVim#1934] [Q&A] [answered]
I would like to set `keyword_length` on the buffer source of nvim-cmp, but I am somehow not able to figure out whats the best way to do it.
Tried the following as a copy paste from some sources that I found but it does not work:
```
return {
  {
    "hrsh7th/nvim-cmp",
    opts = function(_, opts)
      local cmp = require("cmp")
      opts.sources = cmp.config.sources({
        { name = "nvim_lsp" },
        { name = "luasnip" },
        { name = "path" },
        { name = "buffer",
          option = {
            keyword_length = 5,
          },
        },
      })
    end,
  },
}
```

A second thing that I would like to configure is disabling all snippets but the once that are explicitly enabled.
Normally that could be done like this `require("luasnip.loaders.from_vscode").lazy_load({ include = { "python" } })`
How would this be best injected into lazyvim?

Thanks for any tips or help on this :)

**Comment by jyuan0** [comment, marked as answer]
Judging by the nvim-cmp docs, `keyword_length` should be a top-level key (not in `option`). You could try:
```
{ name = "buffer", keyword_length = 5 },
```

For the second thing, I think you could add that code to a friendly-snippets `config` function. Pretty sure there are examples in the answered/closed discussions.

**Reply by mstinsky** [reply]
:man_facepalming: yikes I completely overlooked that. You are right. Thanks!
